src/subscription_manager/repofile.py

```diff
@@ -444,6 +446,9 @@ def fix_content(self, content):
             apt_cont['Suites'] = 'default'
             apt_cont['Components'] = 'all'
             apt_cont['Trusted'] = 'yes'
+            if len(content.arches) > 0 and content.arches[0] != 'ALL':
+                apt_cont['Architectures'] = " ".join(content.arches)
```

Comment:
I can't see what it purpose of this block, but it breaks generating of rhsm.sources. When this block of code is disabled, then rhsm.sources is possible to generate.

Reply:
The target of this is, to specify for which architecture the repo data should be downloaded. 

'Architectures (arch) is a multivalue option defining for which architectures information should be downloaded. If this option isn't set the default is all architectures as defined by the APT::Architectures config option.'

See:
https://manpages.debian.org/buster/apt/sources.list.5.en.html